usage/src/com/cloud/usage/UsageSanityChecker.java

```diff
@@ -72,7 +72,9 @@ protected boolean checkItemCountByPstmt(CheckCase checkCase) throws SQLException
         try (PreparedStatement pstmt = conn.prepareStatement(checkCase.sqlTemplate)) {
             if(checkCase.checkId) {
                 pstmt.setInt(1, lastId);
-                pstmt.setInt(2, maxId);
+                if (maxId > 0) {
+                    pstmt.setInt(2, maxId);
+                }
```

Comment:
This ``if`` creates a circumstance where the second parameter of the ``PreparedStatement`` will not be set.  In this circumstance, an exception will be thrown.  Therefore, this code either always set the value or throw an ``IllegalStateException`` explaining the condition.